src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -118,6 +120,44 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
       else Iterator.empty.next()
   }
 
+  /*
+   * Implemented by means of a buffer to keep track of the last n elements during iteration.
+   */
+  def takeRight(n: Int): Iterator[A] = {
+    if (n == 0) Iterator.empty
```

Comment:
The current collections support negative values for `n`. We should change this test to `if (n <= 0)` to support negative values as well.